indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx

```diff
@@ -0,0 +1,171 @@
+// LICENSE file for more details.
+
+import dashboardURL from 'indico-url:event_editing.dashboard';
+// import manageEditableTypeListURL from 'indico-url:event_editing.manage_editable_type_list'; Uncomment when we have backend ready
+
+import React, {useState} from 'react';
+import {useParams} from 'react-router-dom';
+import {Button, Icon, Checkbox, Message, Search} from 'semantic-ui-react';
+import {Column, Table, SortDirection, WindowScroller} from 'react-virtualized';
+import _ from 'lodash';
+import {
+  TooltipIfTruncated,
+  ManagementPageSubTitle,
+  ManagementPageBackButton,
+} from 'indico/react/components';
+import {useNumericParam} from 'indico/react/util/routing';
+import {Translate} from 'indico/react/i18n';
+import {EditableType} from '../../models';
+import StateIndicator from '../../editing/timeline/StateIndicator';
+import './EditableList.module.scss';
+
+export default function EditableList() {
+  const list = []; // Replace it with list from server
+  const eventId = useNumericParam('confId');
+  const {type} = useParams();
+  const [sortBy, setSortBy] = useState('id');
+  const [sortDirection, setSortDirection] = useState('ASC');
+  const [sortedList, setSortedList] = useState(list); // list from server
+  const [checked, setChecked] = useState([]);
+  const title = {
+    [EditableType.paper]: Translate.string('List of papers'),
+    [EditableType.slides]: Translate.string('List of slides'),
+    [EditableType.poster]: Translate.string('List of posters'),
+  }[type];
+  const columnHeaders = [
+    ['id', Translate.string('ID'), 60],
+    ['code', Translate.string('Code'), 80],
+    ['title', Translate.string('Title'), 600],
+    ['status', Translate.string('Status'), 150],
+    ['assigned', Translate.string('Assigned'), 400],
+  ];
+
+  // eslint-disable-next-line no-shadow
+  const _sortList = ({sortBy, sortDirection}) => {
+    const newList = _.sortBy(list, [sortBy]);
+    if (sortDirection === SortDirection.DESC) {
+      newList.reverse();
+    }
+    return newList;
+  };
+
+  // eslint-disable-next-line no-shadow
+  const _sort = ({sortBy, sortDirection}) => {
+    setSortBy(sortBy);
+    setSortDirection(sortDirection);
+    setSortedList(_sortList({sortBy, sortDirection}));
+  };
+
+  // eslint-disable-next-line react/prop-types
+  const _cellRenderer = ({dataKey, rowIndex}) => {
+    return (
+      <>
+        {dataKey === 'assigned' && <Icon name="user outline" />}
+        <TooltipIfTruncated>
+          <div index={rowIndex} styleName="rowcolumn-tooltip" role="gridcell">
+            {dataKey === 'status' ? (
+              <StateIndicator index={rowIndex} state="needs_submitter_changes" circular label />
+            ) : (
+              sortedList[rowIndex][dataKey]
+            )}
+          </div>
+        </TooltipIfTruncated>
+      </>
+    );
+  };
+
+  const _onChangeHeaderCheckbox = data => {
+    data.checked ? setChecked(sortedList.map(row => row.id)) : setChecked([]);
```

Comment:
Please do not use the ternary operator as a replacement for normal if/else. Basically, don't use the ternary operator unless you do something with the return value of the expression.

Reply:
oh okay, I will change it :+1: